src/actions/pause/extra.js

```diff
@@ -7,19 +7,35 @@
 import { getSymbols, getSource, getSelectedFrame } from "../../selectors";
 import { isReactComponent, isImmutable } from "../../utils/preview";
 import { findClosestClass } from "../../utils/ast";
+import { merge } from "lodash";
 
 import type { ThunkArgs } from "../types";
 
 async function getReactProps(evaluate) {
-  const reactDisplayName = await evaluate(
-    "this.hasOwnProperty('_reactInternalFiber') ? " +
-      "this._reactInternalFiber.type.name : " +
-      "this._reactInternalInstance.getName()"
+  const componentNames = await evaluate(
+    `
+    if(this.hasOwnProperty('_reactInternalFiber')) {
+      let componentNames = []; 
+      let componentNode = this._reactInternalFiber; 
+      while(componentNode) { 
+        componentNames.push(componentNode.type.name); 
+        componentNode = componentNode._debugOwner
+      }
+      componentNames;
+    }
+    else {
+      [this._reactInternalInstance.getName()];
+    }
+    `
   );
-
-  return {
-    displayName: reactDisplayName.result
-  };
+  const items =
+    componentNames.result.preview && componentNames.result.preview.items;
+  if (items) {
+    return {
+      displayName: items[0],
```

Comment:
hmm, i'd prefer not overriding displayName. The rationale is if you are debugging a site that is minified, but has source maps looking up the class name from the symbols table is not minified. Crazy, right?!?!

Reply:
Makes sense! Fixed in 106eeaa17ebc51250d1927ad053372968c2cbacc

Reply:
actually wait 106eeaa still changes the behavior... bc60352db7980803b7cded66a2063a30e3a99155 should do the job.